src/friendly-iframe-embed.js

```diff
@@ -580,6 +618,40 @@ export class FriendlyIframeEmbed {
   }
 }
 
+/**
+ * @param {!Window} win
+ * @param {!Element} ampAdParent
+ * @param {!Element} ampLightbox
+ */
+function renderCloseButtonHeader(win, ampAdParent, ampLightbox) {
+  const el = htmlFor(ampAdParent)`
+    <i-amphtml-ad-close-header role=button>
+      <div>Ad</div>
+      <i-amphtml-ad-close-button class="amp-ad-close-button" role=button>
+      </i-amphtml-ad-close-button>
+    <i-amphtml-ad-close-header>`;
+
+  listenOnce(el, 'click', () => {
+    triggerLightboxClose(win, ampLightbox, /* caller */ ampAdParent);
+    removeElement(el);
+  });
+
+  return el;
+}
+
+/**
+ * @param {!Window} win
+ * @param {!Element} target An amp-lightbox target.
+ * @param {!Element} caller Whomever.
+ */
+function triggerLightboxClose(win, target, caller) {
+  const event = createCustomEvent(win, 'tap', /* detail */ {});
```

Comment:
I don't know all the nuances, is there no better way to do this? like querying and installing event listener on `i-amphtml-ad-close-button`?

Reply:
Getting the component implementation is not possible given that `amp-lightbox` may not be loaded on the host frame.

Reply:
(And a listener is no good as we need to push the event down, not bubble it up).

Reply:
i see, and action icon and lightbox are in two different frames.